src/ocm/components/clusterConfiguration/manifestsConfiguration/components/CustomManifestForm.tsx

```diff
@@ -65,9 +69,43 @@ export const CustomManifestForm = <CustomManifestFormValues extends FormViewMani
                 customManifest.yamlContent === manifest.manifestYaml,
             ),
         );
+        if (cluster && manifestsModified.length > 0) {
+          //See if manifests exists previously to make a patch or create a new one
+          const manifestsThatExists = manifestsModified.filter(
+            (manifest) => manifest.fakeId !== '',
+          );
+          const newManifestsToCreate = manifestsModified.filter(
+            (manifest) => manifest.fakeId === '',
+          );
+          await Promise.all(
+            manifestsThatExists.map(async (updatedManifest) => {
+              const existingFolderByFakeId = updatedManifest.fakeId?.substring(
+                0,
+                updatedManifest.fakeId.indexOf('#'),
+              );
+              const existingFolder: FormViewManifestFolder =
+                existingFolderByFakeId === 'manifests' || existingFolderByFakeId === 'openshift'
+                  ? existingFolderByFakeId
+                  : 'manifests';
+              const existingFileName = updatedManifest.fakeId?.substring(
+                updatedManifest.fakeId.indexOf('#') + 1,
+              );
+              const existingManifest: FormViewManifest = {
+                folder: existingFolder,
+                filename: existingFileName || '',
+                manifestYaml: '',
+              };
+              await ClustersService.updateCustomManifest(
```

Comment:
You should not use `await` in the array items of a `Promise.all`, as it would be making each individual promise have to wait to the previous before it. More details here:
https://stackoverflow.com/questions/64515055/should-i-use-await-inside-promise-all

The purpose of `Promise.all` is to launch all of the promises at the same time, and resolve them as they happen.